synapse/lib/stormlib/cache.py

```diff
@@ -0,0 +1,175 @@
+class FixedCache(s_stormtypes.StormType):
+    '''
+    A StormLib API instance of a Storm Fixed Cache.
+    '''
+    _storm_locals = (
+        {'name': 'query', 'desc': 'Get the callback Storm query as string.',
+         'type': {'type': 'gtor', '_gtorfunc': '_gtorQuery',
+                  'returns': {'type': 'str', 'desc': 'The callback Storm query text.', }}},
+        {'name': 'get', 'desc': 'Get an item from the cache by key.',
+         'type': {'type': 'function', '_funcname': '_methGet',
+                  'args': (
+                      {'name': 'key', 'type': 'any', 'desc': 'The key to lookup.'},
+                  ),
+                  'returns': {'type': 'any',
+                              'desc': 'The value from the cache, or the callback query if it does not exist', }}},
+        {'name': 'pop', 'desc': 'Pop an item from the cache.',
+         'type': {'type': 'function', '_funcname': '_methPop',
+                  'args': (
+                      {'name': 'key', 'type': 'any', 'desc': 'The key to pop.'},
+                  ),
+                  'returns': {'type': 'any',
+                              'desc': 'The value from the cache, or $lib.null if it does not exist', }}},
+        {'name': 'put', 'desc': 'Put an item into the cache.',
+         'type': {'type': 'function', '_funcname': '_methPut',
+                  'args': (
+                      {'name': 'key', 'type': 'any', 'desc': 'The key put in the cache.'},
+                      {'name': 'value', 'type': 'any', 'desc': 'The value to assign to the key.'},
+                  ),
+                  'returns': {'type': 'null', }}},
+        {'name': 'clear', 'desc': 'Clear all items from the cache.',
+         'type': {'type': 'function', '_funcname': '_methClear',
+                  'returns': {'type': 'null', }}},
+    )
+    _storm_typename = 'cache:fixed'
+    _ismutable = False
+
+    def __init__(self, runt, query, size=CACHE_SIZE_DEFAULT):
+        s_stormtypes.StormType.__init__(self)
+        self.runt = runt
+        self.size = size
+        self.query = query
+        self.locls.update(self.getObjLocals())
+        self.gtors.update({
+            'query': self._gtorQuery,
+        })
+
+        self.cache = s_cache.FixedCache(self._runCallback, size=size)
+
+    def __len__(self):
+        return len(self.cache)
+
+    async def stormrepr(self):
+        if len(qtext := self.query.text) > 100:
+            qtext = qtext[:100] + '...'
+        return f'{self._storm_typename}: size={self.size} query="{qtext}"'
+
+    def getObjLocals(self):
+        return {
+            'pop': self._methPop,
+            'put': self._methPut,
+            'get': self._methGet,
+            'clear': self._methClear,
+        }
+
+    async def _gtorQuery(self):
+        return self.query.text
+
+    async def _runCallback(self, key):
+        opts = {'vars': {'cache_key': key}}
+        async with self.runt.getSubRuntime(self.query, opts=opts) as subr:
```

Comment:
Should this runtime be able to update var values in the parent runtime? I think we might want to protect against that and make this a CmdRuntime instead of SubRuntime.

Reply:
I was thinking it should, since I was looking at the scoping for the Query and Pipe objects as comparable patterns.  Probably a goback.